ext/rugged/rugged_repo.c

```diff
@@ -1058,59 +1058,64 @@ static VALUE rb_git_repo_expand_oids(int argc, VALUE *argv, VALUE self)
 {
 	VALUE rb_result, rb_oids, rb_expected_type;
 
-	git_otype expected_type = GIT_OBJ_ANY;
-
 	git_repository *repo;
-	git_oid oid;
 	git_odb *odb;
-	int i, error;
+	git_odb_expand_id *expand;
+	long i, expand_count;
+	int error;
 
 	Data_Get_Struct(self, git_repository, repo);
-
 	rb_scan_args(argc, argv, "11", &rb_oids, &rb_expected_type);
 
 	Check_Type(rb_oids, T_ARRAY);
-	expected_type = rugged_otype_get(rb_expected_type);
+	expand_count = RARRAY_LEN(rb_oids);
+	expand = alloca(expand_count * sizeof(git_odb_expand_id));
 
-	error = git_repository_odb(&odb, repo);
-	rugged_exception_check(error);
+	for (i = 0; i < expand_count; ++i) {
+		VALUE rb_hex = rb_ary_entry(rb_oids, i);
+		Check_Type(rb_hex, T_STRING);
 
-	rb_result = rb_hash_new();
+		rugged_exception_check(
+			git_oid_fromstrn(&expand[i].id, RSTRING_PTR(rb_hex), RSTRING_LEN(rb_hex))
+		);
+		expand[i].length = RSTRING_LEN(rb_hex);
+	}
 
-	for (i = 0; i < RARRAY_LEN(rb_oids); ++i) {
-		VALUE hex_oid = rb_ary_entry(rb_oids, i);
-		git_oid found_oid;
+	if (TYPE(rb_expected_type) == T_ARRAY) {
```

Comment:
Wow aren't you fancy.  Do you want to update the documentation to reflect that you can now pass this fancy new array of types?